src/main.rs

```diff
@@ -75,53 +63,35 @@ struct Newer {
 
 impl Newer {
     fn new(args: Vec<String>) -> Result<Newer, Error> {
-        enum State {
-            Output,
-            Input,
-            Command,
-        }
+        let mut args = args.iter();
+
+        let newest = args
+            .by_ref()
+            .take_while(|&a| a != ":")
+            .map(|arg| match find_newest(arg) {
+                Err(e) if e.kind() == ErrorKind::NotFound => Ok(SystemTime::UNIX_EPOCH),
+                n @ _ => n,
```

Comment:
Just when you think Rust can't throw more magical shit at you, it surprises you.

Reply:
The irony is that it's the same length as `Ok(n) => n`!

Reply:
It's different. `n @ _ => n` matches all values. In this case, it replaces:

```rust
Err(e) => e,
Ok(v) => v,
```